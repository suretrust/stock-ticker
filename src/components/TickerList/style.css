.ticker-list {
    width: 440px;
    box-sizing: border-box;
    background: #FFFFFF;
    box-shadow: 0px 10px 20px rgba(0, 38, 76, 0.1);
    border-radius: 8px;
    margin: 0 auto;
    padding: 0 16px;
    max-height: 490px;
    overflow-y: scroll;
}


.ticker {
    font-weight: 500;
    font-size: 15px;
    line-height: 24px;
    color: #00264C;
    height: 88px;
    display: flex;
    text-align: left;
    align-items: center;
    border-bottom: 1px solid #EDF4F7;
    width: 100%;
    cursor: pointer;
}