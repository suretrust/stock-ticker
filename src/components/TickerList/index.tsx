import React from 'react';
import './style.css';

interface TickerListProps {
  allTickers: any[];
  handleTickerClick: (symbol: string) => void;
}

interface Ticker {
  ticker: string;
  name: string;
}

const TickerList: React.FC<TickerListProps> = ({
  allTickers,
  handleTickerClick,
}) => {
  if (!allTickers?.length) return null;

  return (
    <div className='ticker-list'>
      {allTickers.map((ticker: Ticker) => (
        <div
          className='ticker'
          key={ticker.name}
          onClick={() => handleTickerClick(ticker.ticker)}
        >
          <div>{ticker.name}</div>
        </div>
      ))}
    </div>
  );
};

export default TickerList;
