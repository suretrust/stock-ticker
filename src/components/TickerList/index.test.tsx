import { render, screen, fireEvent } from '@testing-library/react';
import TickerList from '.';

const allTickers = [
  {
    name: 'test name',
    ticker: 'TEST',
  },
  {
    name: 'test name 2',
    ticker: 'TEST',
  },
];

const handleTickerClick = jest.fn();

describe('TickerList component', () => {
  it('renders TickerList as fragment', () => {
    const { asFragment } = render(
      <TickerList
        allTickers={allTickers}
        handleTickerClick={handleTickerClick}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders first ticker name from list of all tickers', () => {
    render(
      <TickerList
        allTickers={allTickers}
        handleTickerClick={handleTickerClick}
      />
    );
    const tickerName = screen.getByText(allTickers[0].name);

    expect(tickerName).toBeTruthy();
  });

  it('renders second ticker name from list of all tickers', () => {
    render(
      <TickerList
        allTickers={allTickers}
        handleTickerClick={handleTickerClick}
      />
    );
    const tickerName = screen.getByText(allTickers[1].name);

    expect(tickerName).toBeTruthy();
  });

  it('calls handleTickerClick on clicking first ticker name', () => {
    render(
      <TickerList
        allTickers={allTickers}
        handleTickerClick={handleTickerClick}
      />
    );
    const tickerName = screen.getByText(allTickers[0].name);
    fireEvent.click(tickerName);

    expect(handleTickerClick).toHaveBeenCalledTimes(1);
    expect(handleTickerClick).toHaveBeenCalledWith(allTickers[0].ticker);
  });

  it('calls handleTickerClick with ticker on clicking second ticker name', () => {
    render(
      <TickerList
        allTickers={allTickers}
        handleTickerClick={handleTickerClick}
      />
    );
    const tickerName = screen.getByText(allTickers[1].name);
    fireEvent.click(tickerName);

    expect(handleTickerClick).toHaveBeenCalledWith(allTickers[1].ticker);
  });
});
