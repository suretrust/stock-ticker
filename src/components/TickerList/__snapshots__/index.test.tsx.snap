// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickerList component renders TickerList as fragment 1`] = `
<DocumentFragment>
  <div
    class="ticker-list"
  >
    <div
      class="ticker"
    >
      <div>
        test name
      </div>
    </div>
    <div
      class="ticker"
    >
      <div>
        test name 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
